File: src/Client/Logistics.Client/Models/UpdateLoad.cs
using Logistics.Models;

namespace Logistics.Client.Models;

public record UpdateLoad
{
    public string? Id { get; set; }
    public string? Name { get; set; }
    public string? SourceAddress { get; set; }
    public string? DestinationAddress { get; set; }
    public double? DeliveryCost { get; set; }
    public double? Distance { get; set; }
    public string? AssignedDispatcherId { get; set; }
    public string? AssignedDriverId { get; set; }
    public LoadStatusDto? Status { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/models/updateLoad.ts
import {LoadStatus} from '@core/enums';

export interface UpdateLoad {
  id: string;
  name?: string;
  originAddress: string;
  originAddressLat: number;
  originAddressLong: number;
  destinationAddress: string;
  destinationAddressLat: number;
  destinationAddressLong: number;
  deliveryCost: number;
  distance: number;
  assignedDispatcherId: string;
  assignedTruckId: string;
  status?: LoadStatus;
}
